cypher-shell/src/main/java/org/neo4j/shell/prettyprint/OutputFormatter.java

```diff
@@ -57,70 +52,88 @@ String formatValue(@Nonnull final Value value) {
         }
     }
 
-    private String pathAsString(Path path) {
-        List<String> list = new LinkedList<>();
+    @Nonnull
+    default String pathAsString(@Nonnull Path path) {
+        List<String> list = new ArrayList<>(path.length());
         Node lastTraversed = path.start();
         if (lastTraversed != null) {
             list.add(nodeAsString(lastTraversed));
-        }
 
-        for (Path.Segment segment : path) {
-            Relationship relationship = segment.relationship();
-            if (relationship.startNodeId() == lastTraversed.id()) {
-                //-[:r]->
-                list.add("-" + relationshipAsString(relationship) + "->");
-                list.add(nodeAsString(segment.end()));
-                lastTraversed = segment.start();
-            } else {
-                list.add("<-" + relationshipAsString(relationship) + "-");
-                list.add(nodeAsString(segment.end()));
-                lastTraversed = segment.end();
+            for (Path.Segment segment : path) {
+                Relationship relationship = segment.relationship();
+                if (relationship.startNodeId() == lastTraversed.id()) {
+                    //-[:r]->
+                    list.add("-" + relationshipAsString(relationship) + "->");
+                    list.add(nodeAsString(segment.end()));
+                    lastTraversed = segment.start();
+                } else {
+                    list.add("<-" + relationshipAsString(relationship) + "-");
+                    list.add(nodeAsString(segment.end()));
+                    lastTraversed = segment.end();
+                }
             }
         }
 
         return list.stream().collect(Collectors.joining());
     }
 
-    private String listAsString(List<String> list) {
-        return list.stream().collect(Collectors.joining(COMMA_SEPARATOR,"[","]"));
-    }
-
-    private String mapAsString(Map<String, Object> map) {
-        if (map.isEmpty()) {
-            return "";
-        }
-        return map.entrySet().stream()
-                        .map(e -> escape(e.getKey()) + COLON_SEPARATOR + e.getValue())
-                        .collect(Collectors.joining(COMMA_SEPARATOR,"{","}"));
-    }
-
-    private String relationshipAsString(Relationship relationship) {
+    @Nonnull default String relationshipAsString(@Nonnull Relationship relationship) {
         List<String> relationshipAsString = new ArrayList<>();
         relationshipAsString.add(COLON + escape(relationship.type()));
         relationshipAsString.add(mapAsString(relationship.asMap(this::formatValue)));
 
         return "[" + joinWithSpace(relationshipAsString) + "]";
     }
 
-    private String nodeAsString(@Nonnull final Node node) {
+    @Nonnull default String nodeAsString(@Nonnull final Node node) {
```

Comment:
How come `nodeAsString` and `relationshipAsString` are not static like the rest of the to-string functions? Is there a special case I'm missing?